pvacseq/lib/vaccine_design.py

```diff
@@ -158,6 +159,221 @@ def anneal(self):
+
+#draw perpindicular line to arc to mark junction
+def draw_junction_w_label(junct_score, t, pen_thin, angle):
+    reset = t.heading()
+    t.rt(90)
+    t.pencolor("black")
+    t.pensize(pen_thin)
+    t.forward(10)
+    t.back(20)
+    t.pu()
+    if (angle >= 0 and angle < 70):
+        write_junct_score(t, junct_score, 15)
+    elif (angle >= 65 and angle < 115):
+        write_junct_score(t, junct_score, 10)
+    elif (angle >= 115 and angle < 165):
+        write_junct_score(t, junct_score, 20)
+    elif (angle >= 165 and angle < 195):
+        write_junct_score(t, junct_score, 25)
+    elif (angle >= 195 and angle < 245):
+        write_junct_score(t, junct_score, 35)
+    elif (angle >= 245 and angle < 295):
+        write_junct_score(t, junct_score, 15)
+    else:
+        write_junct_score(t, junct_score, 20)
+    t.pd()
+    t.forward(10)
+    t.setheading(reset)
+    return()
+
+#draw second line of junctions with amino acid additions
+def draw_junction(t, pen_thin):
+    reset = t.heading()
+    t.rt(90)
+    t.pencolor("black")
+    t.pensize(pen_thin)
+    t.forward(10)
+    t.back(20)
+    t.forward(10)
+    t.setheading(reset)
+    return()
+
+#draw arc for peptide
+def draw_arc_peptide(peptide, length, count, angle, t, circle_radius, conversion_factor, pep_id_space):
+    t.pencolor(get_color(count))
+    t.circle(circle_radius, (conversion_factor * length) / 2)
+    t.pu()
+    reset = t.heading()
+    t.left(90)
+    t.forward(pep_id_space)
+    if (angle > 80 and angle < 100) or (angle > 260 and angle < 280):
+        t.write(peptide, align="center", font=("Arial", 10, "bold"))
+    elif (angle > 0 and angle < 90) or (angle > 270 and angle < 360):
+        t.write(peptide, align="right", font=("Arial", 10, "bold"))
+    else:
+        t.write(peptide, align="left", font=("Arial", 10, "bold"))
+    t.back(pep_id_space)
+    t.setheading(reset)
+    t.pd()
+    t.circle(circle_radius, (conversion_factor * length) / 2)
+
+#draw arc for amino acid inserts to junctions
+def draw_arc_junct(peptide, length, t, conversion_factor, junct_seq_space, circle_radius):
+    #t.pencolor("black")
+    t.circle(circle_radius, (conversion_factor * length) / 2)
+    t.pu()
+    reset = t.heading()
+    t.left(90)
+    t.back(junct_seq_space)
+    t.write(peptide, align="center")
+    t.forward(junct_seq_space)
+    t.setheading(reset)
+    t.pd()
+    t.circle(circle_radius, (conversion_factor * length) / 2)
+
+#print turtle screen to a postscript file, convert to pdf
+def output_screen(t, out_f):
+    ps_file = "/".join((out_f, "vaccine.ps"))
+    out_file = "/".join((out_f, "vaccine.jpg"))
+    ts = t.getscreen()
+    ts.getcanvas().postscript(file=ps_file)
+    os.system('convert -density 300 -quality 100 ' + ps_file + " " + out_file)
+    os.system('rm ' + ps_file)
+    return()
+
+def output_vaccine_png(input_file, out_f):
+    min_pep_length = 8
+    max_pep_length = 100
+
+    pep_seqs = []
+    with open(input_file, 'r') as input_f:
```

Comment:
I would make the parsing of the input file into its own method that returns `pep_seq` and `junct_scores`.